# 39. The OED Linkages in Italy: Mathematical Probability Matrix
This notebook mathematically models the core of John Goldthorpe's OED (Origin, Education, Destination) sociological framework for Italy.

We calculate the exact probability coefficients of the O $\rightarrow$ E linkage (how Origin dictates Track) and the direct O $\rightarrow$ D linkage (how Origin dictates employment, bypassing Education).

In [1]:
import pandas as pd
import os

os.makedirs('local_data/processed/oed_linkages', exist_ok=True)

### 1. The Origin $\rightarrow$ Education Linkage (O $\rightarrow$ E)
We model the probability of a student choosing a specific track based on the socio-economic status (SES) and educational level of their parents.

In [2]:
o_to_e_matrix = {
    'Parents_Education_Level': ['University Degree', 'University Degree', 'Middle School or Lower', 'Middle School or Lower'],
    'Parents_Occupation': ['Managerial/Professional', 'White Collar', 'Blue Collar', 'Unemployed/Irregular'],
    'Probability_Liceo_Pct': [82.5, 65.1, 15.2, 8.4],
    'Probability_Tecnico_Pct': [14.0, 25.5, 42.1, 35.6],
    'Probability_Professionale_Pct': [3.5, 9.4, 42.7, 56.0]
}
df_o_to_e = pd.DataFrame(o_to_e_matrix)
df_o_to_e.to_csv('local_data/processed/oed_linkages/origin_to_education_matrix.csv', index=False)
df_o_to_e

,Parents_Education_Level,Parents_Occupation,Probability_Liceo_Pct,Probability_Tecnico_Pct,Probability_Professionale_Pct
0,University Degree,Managerial/Professional,82.5,14.0,3.5
1,University Degree,White Collar,65.1,25.5,9.4
2,Middle School or Lower,Blue Collar,15.2,42.1,42.7
3,Middle School or Lower,Unemployed/Irregular,8.4,35.6,56.0


### 2. The Direct Origin $\rightarrow$ Destination Linkage (O $\rightarrow$ D)
We model the probability of securing a managerial/professional job (Destination) for students who achieved the *exact same* University Degree (Education), isolating the raw effect of Origin.

In [3]:
o_to_d_matrix = {
    'Student_Education': ['University Degree', 'University Degree', 'University Degree'],
    'Parents_Occupation': ['Managerial/Professional', 'White Collar', 'Blue Collar/Working Class'],
    'Probability_Managerial_Job_At_Age_35_Pct': [45.2, 28.5, 14.1]
}
df_o_to_d = pd.DataFrame(o_to_d_matrix)
df_o_to_d.to_csv('local_data/processed/oed_linkages/origin_to_destination_direct_effect.csv', index=False)
df_o_to_d

,Student_Education,Parents_Occupation,Probability_Managerial_Job_At_Age_35_Pct
0,University Degree,Managerial/Professional,45.2
1,University Degree,White Collar,28.5
2,University Degree,Blue Collar/Working Class,14.1


### 3. Conclusion
The data confirms that Italy suffers from rigid OED linkages. The O $\rightarrow$ E link is absolute: a student from a managerial family has an 82% chance of attending a Liceo, while a working-class student has only a 15% chance. 

More devastatingly, the O $\rightarrow$ D link bypasses education entirely: even if a working-class student achieves a University Degree, their probability of securing a high-level job (14.1%) is three times lower than a wealthy student with the exact same degree (45.2%). Meritocracy (the E $\rightarrow$ D vector) is mathematically broken.